use-cases/using-Intel-QAT-and-kata.md

```diff
@@ -0,0 +1,491 @@
+the container.
+```
+bash-5.0# ls /etc
+c6xxvf_dev0.conf   c6xxvf_dev11.conf  c6xxvf_dev14.conf  c6xxvf_dev3.conf  c6xxvf_dev6.conf  c6xxvf_dev9.conf  resolv.conf
+c6xxvf_dev1.conf   c6xxvf_dev12.conf  c6xxvf_dev15.conf  c6xxvf_dev4.conf  c6xxvf_dev7.conf  hostname
+c6xxvf_dev10.conf  c6xxvf_dev13.conf  c6xxvf_dev2.conf   c6xxvf_dev5.conf c6xxvf_dev8.conf  hosts
+```
+
+* Check dmesg inside the container to see if there are any issues with the 
+Intel QAT driver
+
+* If there are issues building the OpenSSL Intel QAT container image then 
+check to make sure that runc is the default runtime for building container.
+```
+$ cat /etc/systemd/system/docker.service.d/50-runtime.conf
+[Service]
+Environment="DOCKER_DEFAULT_RUNTIME=--default-runtime runc"
+```
+
+## Optional Scripts
+
+### Verify Intel QAT card counters are incremented
+
+Use the lspci command to figure out which PCI bus the Intel QAT accelerators 
+are on. When the accelerator is actually used, the counters will increment. 
+This is an easy way to verify the Intel QAT is being used by the containerized 
+application. In these instructions, only the first physical function and its 
+virtual functions were exposed.
+
+```
+$ for i in 0434 0435 37c8 1f18 1f19; do lspci -d 8086:$i; done
+b1:00.0 Co-processor: Intel Corporation C62x Chipset QuickAssist Technology (rev 04)
+	b3:00.0 Co-processor: Intel Corporation C62x Chipset QuickAssist Technology (rev 04)
+	b5:00.0 Co-processor: Intel Corporation C62x Chipset QuickAssist Technology (rev 04
+$ sudo watch cat /sys/kernel/debug/qat_c6xx_0000\:b1\:00.0/fw_counters
+$ sudo watch cat /sys/kernel/debug/qat_c6xx_0000\:b3\:00.0/fw_counters
+$ sudo watch cat /sys/kernel/debug/qat_c6xx_0000\:b5\:00.0/fw_counters
+```
+
+### Unbind all VF’s from VFIO driver, reset, and rebind to the host
+
+```
+# Loop through all the virtual functions and unbind the VFIO driver (OPTIONAL TO RESET)
+for f in /sys/bus/pci/devices/0000:$QAT_PCI_BUS_PF_1/virtfn*; do QAT_PCI_BUS_VF=$(basename $(readlink $f)); echo $QAT_PCI_BUS_VF | sudo tee --append /sys/bus/pci/drivers/VFIO-pci/unbind; echo $QAT_PCI_BUS_VF | sudo tee --append /sys/bus/pci/drivers/c6xxvf/bind; done
+
+QAT_PCI_BUS_PF_NUMBERS=$(lspci | grep Quick | cut -d ' ' -f 1)
+QAT_PCI_BUS_PF_1=$(echo $QAT_PCI_BUS_PF_NUMBERS | cut -d ' ' -f 1)
+echo 16 | sudo tee /sys/bus/pci/devices/0000:$QAT_PCI_BUS_PF_1/sriov_numvfs
+QAT_PCI_ID_VF=$(cat /sys/bus/pci/devices/0000:${QAT_PCI_BUS_PF_1}/virtfn0/uevent | grep PCI_ID)
+QAT_VENDOR_AND_ID_VF=$(echo ${QAT_PCI_ID_VF/PCI_ID=} | sed 's/:/ /')
+echo $QAT_VENDOR_AND_ID_VF | sudo tee --append /sys/bus/pci/drivers/VFIO-pci/new_id
+
+###Loop through all the virtual functions and bind to the VFIO driver
+for f in /sys/bus/pci/devices/0000:$QAT_PCI_BUS_PF_1/virtfn*; do QAT_PCI_BUS_VF=$(basename $(readlink $f)); echo $QAT_PCI_BUS_VF | sudo tee --append /sys/bus/pci/drivers/c6xxvf/unbind; echo $QAT_PCI_BUS_VF | sudo tee --append /sys/bus/pci/drivers/VFIO-pci/bind; done
```

Comment:
Missing ` ```sh ` code block around this chunk of shell. I appreciate you've crunched it into a single line to make it easy to copy/paste but the downside of doing this:

- It's very hard to read (and I'd argue users *do* care about commands we're asking them to blindly paste into a terminal).
- It makes diffs bigger than necessary if this line subsequently changes.

My advice would be to break it across multiple lines as clarity always trumps convenience imho ;)

Reply:
No problem. The next rev will have this broken into multiple lines.